org-static-blog.el

```diff
@@ -312,7 +318,37 @@ e.g. `(('foo' 'file1.org' 'file2.org') ('bar' 'file2.org'))`"
               (push post-filename (cdr (assoc-string tag tag-tree t)))
             (push (cons tag (list post-filename)) tag-tree)))))
     tag-tree))
-
+(defun org-static-blog-get-preview (post-filename)
+  (with-temp-buffer
+    (insert-file-contents (org-static-blog-matching-publish-filename post-filename))
+    (let ((title-start)
+          (paragraph-end)
+          (post-start)
+          (post-end))
+      (goto-char (point-min))
+      (setq title-start (search-forward "<div id=\"content\">"))
+      (search-forward "<h1 class=\"post-title\">")
+      (replace-match "<h2 class=\"post-title\">")
```

Comment:
Why are `h1` replaced with `h2`? This probably breaks if the post contains internal `h2`s. I think we should leave the headline levels as they are.

Reply:
I guess I might be able to answer this one...

`h1` is usually used for page titles and supposed to be only one on any given page. This help search engines and screen readers understand what is the title of the page. 

We have a SEO/HTML problem in general where index page contains a lot of `h1` tag (one for each post).

I was trying to come up with an elegant solution to this problem before, but unfortunately still don't have any suggestions on how to fix it without lots of search/replaces.

In this particular example it might be safe to assume that we will never get any other `h*` tags since we are cutting out only first paragraph.

Reply:
True, any internal `h2`s would be hidden anyway.

Do you know off-hand how other blogging engines do this kind of thing?

Reply:
I my opinion, the source of this issue is in the fact that most of the time operations are done over already compiled HTML source code. This leaves us to deal with any structural modifications with string operations over HTML code.
Other org-mode engines could be getting such meta information from the source `.org` files directly. In this case they can define stricture of the resulting HTML after the gathered the data.

I don't have a good answer to how to solve it for us right at the moment, but I think there should be a nice way out, without relying on shaky string operations.

Reply:
Wordpress indeed demotes H1s to H2s on the index page.

I would say, let's keep them as H1s for now, and open a separate issue for dealing with this not just on this page, but on the current front page as well.

Let's continue this discussion in #44.